### PR TITLE
GUI issues fixed in population atlas

### PR DESCRIPTION
**Pull request by saimarathore**
GUI issues fixed in population atlas
Code improvement and error checks

### CODE REVIEW

File: src/applications/TrainingModule.cpp [hunk 1]
```diff
@@ -2174,7 +2174,8 @@ std::vector<int> TrainingModule::EffectSizeBasedFeatureSelection(const VariableS
     //check crossvalidated performance after adding the current feature
     double bestCV = 0;
     double bestC = 1;
-    double bestG = 1 / currentFeatureSet.Cols();
+    double numerator = 1;
```
Comment (sarthakpati) [inline src/applications/TrainingModule.cpp:2177]: Why is numerator defined as ```1``` if it isn't being used anywhere else?

File: src/view/gui/fMainWindow.cpp [hunk 2]
```diff
@@ -4523,20 +4523,36 @@ void fMainWindow::CallGeneratePopualtionAtlas(const std::string inputFileName, c
...
+    int images_index = -1;
+    int atlas_labels_index = -1;
     for (size_t k = 0; k < allRows[0].size(); k++)
     {
       auto check_wrap = allRows[0][k];
       std::transform(check_wrap.begin(), check_wrap.end(), check_wrap.begin(), ::tolower);
-
       if (check_wrap == "patient_ids")
-        patient_ids.push_back(allRows[j][k]);
+        patient_id_index = k;
```
Comment (sarthakpati) [inline src/view/gui/fMainWindow.cpp:4537]: Trying to understand this change: this seems to indicate that ```patient_id_index``` gets overwritten, is that right?

File: src/view/gui/fMainWindow.cpp [hunk 3]
```diff
@@ -4523,20 +4523,36 @@ void fMainWindow::CallGeneratePopualtionAtlas(const std::string inputFileName, c
...
       else if (check_wrap == "images")
-        image_paths.push_back(allRows[j][k]);
+        images_index = k;
```
Comment (sarthakpati) [inline src/view/gui/fMainWindow.cpp:4539]: Same as previous comment: this seems to indicate that ```images_index``` gets overwritten, is that right?

File: src/applications/TrainingModule.cpp (hunk 1 above)
Comment (saimarathore) [reply src/applications/TrainingModule.cpp:2177]: removed.

File: src/view/gui/fMainWindow.cpp (hunk 2 above)
Comment (saimarathore) [reply src/view/gui/fMainWindow.cpp:4537]: yes. this gets over-written.

File: src/view/gui/fMainWindow.cpp (hunk 3 above)
Comment (saimarathore) [reply src/view/gui/fMainWindow.cpp:4539]: this gets overwritten.

File: src/applications/TrainingModule.cpp (hunk 1 above)
Comment (saimarathore) [reply src/applications/TrainingModule.cpp:2177]: using in line 2178 of TrainingModule.cpp

**Comment by saimarathore** [comment]
this PR fixes the following issues: #1069 and #1062

**Comment by sarthakpati** [comment]
> @sarthakpati this PR fixes the following issues: #1069 and #1062

#1062 is still not working. When I am using the latest sample data, I am getting the following error:

image

Inputs:

image

**Comment by saimarathore** [comment]
paths in labels.csv should be absolute.

**Comment by sarthakpati** [comment]
> @sarthakpati paths in labels.csv should be absolute.

Oh okay. I just used the sample data that you had provided. Should this be mentioned in the [documentation](https://cbica.github.io/CaPTk/Glioblastoma_Atlas.html), then?

**Pull request merged**